### PR TITLE
build: update to Go 1.19

### PR DESCRIPTION
**Pull request by lgfa29**
Update toolchain to use Go 1.19.3.

The failing CircleCI will be fixed once https://github.com/hashicorp/nomad-autoscaler/pull/609 is merged.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -23,11 +23,7 @@ linters:
...
     - misspell
     - govet
-    - deadcode
-    - varcheck
     - ineffassign
-    - structcheck
     - unconvert
     - gofmt
-    - maligned
   fast: false
```
Comment (lgfa29) [inline .golangci.yml:29]: These removed linters are marked as deprecated.

File: GNUmakefile [hunk 2]
```diff
@@ -25,23 +25,23 @@ tools: lint-tools test-tools generate-tools
...
 .PHONY: lint-tools
 lint-tools: ## Install the tools used to lint
 	@echo "==> Installing lint tools..."
-	GO111MODULE=on cd tools && go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.24.0
-	GO111MODULE=on cd tools && go get -u honnef.co/go/tools/cmd/staticcheck@2020.1.6
-	GO111MODULE=on cd tools && go get github.com/hashicorp/go-hclog/hclogvet@v0.1.3
-	GO111MODULE=on cd tools && go get github.com/hashicorp/hcl/v2/cmd/hclfmt@d0c4fa8b0bbc2e4eeccd1ed2a32c2089ed8c5cf1
+	cd tools && go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.50.1
+	cd tools && go install honnef.co/go/tools/cmd/staticcheck@2022.1.3
+	cd tools && go install github.com/hashicorp/go-hclog/hclogvet@v0.1.5
```
Comment (lgfa29) [inline GNUmakefile:43]: The version bumps here were necessary to support the new Go version. I was getting weird errors until they were all upgraded.

File: agent/config/config_test.go [hunk 3]
```diff
@@ -1,7 +1,6 @@
 package config
 
 import (
-	"io/ioutil"
```
Comment (lgfa29) [inline agent/config/config_test.go:4]: `io/ioutil` was deprecated in Go 1.18

**Review by jrasell — APPROVED** [review]
LGTM!

**Review by tgross — APPROVED** [review]

**Pull request merged**